UnityProject/Assets/Scripts/Systems/ChemistryEffects/ChemExplosionPullPush.cs

```diff
@@ -0,0 +1,102 @@
+using System.Collections;
+using UnityEngine;
+using Systems.Explosions;
+using HealthV2;
+using Core;
+using Cysharp.Threading.Tasks;
+using Logs;
+using Systems.Atmospherics;
+using TileManagement;
+using UniversalObjectPhysics = Core.Physics.UniversalObjectPhysics;
+
+namespace Chemistry.Effects
+{
+	[CreateAssetMenu(fileName = "effect", menuName = "ScriptableObjects/Chemistry/Effect/ChemExplosionPullPush")]
+	public class ChemExplosionPullPush : ChemExplosion
+	{
+		[SerializeField, Tooltip("Should this effects force push not pull")] private bool ShouldPush = false;
+		[SerializeField] private string effectName = "Dark Matter";
+
+
+		[SerializeField] private GameObject cataclysmPrefab = null;
+		[SerializeField] private float cataclysmThreshold = 2000;
+
+		public override IEnumerator NowExplosion(MonoBehaviour sender, float amount)
+		{
+			yield return WaitFor.Seconds(Delay);
+
+			UniversalObjectPhysics objectBehaviour = sender.GetComponent<UniversalObjectPhysics>();
+			RegisterObject registerObject = sender.GetComponent<RegisterObject>();
+			BodyPart bodyPart = sender.GetComponent<BodyPart>();
+			ExplosionNode node = ExplosionTypes.NodeTypes[explosionType];
+
+			float strength = ChemistryUtils.CalculateYieldFromReaction(amount, potency);
+
+			bool insideBody = bodyPart == true && bodyPart.HealthMaster == true;
+
+			var picked = sender.GetComponent<Pickupable>();
```

Comment:
Too many get components!!!!! Use CommonComponents instead.

Reply:
Do standard objects have common componenets on them?

Reply:
every prefab that inherits from Object and Item should have it
double check

Reply:
They do, will update these to getcomponentcustom now